piezo_web_app/PiezoWebApp/src/services/kubernetes/kubernetes_adapter.py

```diff
@@ -0,0 +1,62 @@
+import kubernetes
+from kubernetes.client.rest import ApiException
+import requests
+
+from PiezoWebApp.src.services.kubernetes.i_kubernetes_adapter import IKubernetesAdapter
+
+
+API_BASE_URL = 'localhost:8001'
+
+# str | The custom resource's group name
+CRD_GROUP = 'sparkoperator.k8s.io'
+
+# str | The custom resource's plural name. For TPRs this would be lowercase plural kind.
+CRD_PLURAL = 'sparkapplications'
+
+# str | The custom resource's version
+CRD_VERSION = 'v1beta1'
+
+
+class KubernetesAdapter(IKubernetesAdapter):
+    def __init__(self, logger):
+        self._logger = logger
+        config = kubernetes.config.load_incluster_config()
+        api_client = kubernetes.client.ApiClient(config)
+        self._connection = kubernetes.client.CustomObjectsApi(api_client)
+
+    def delete_job(self, job_name, namespace):
+        url = f'http://${API_BASE_URL}/apis/${CRD_GROUP}/${CRD_VERSION}/namespaces/${namespace}/${CRD_PLURAL}/${job_name}'
```

Comment:
We have a choice here to either use the python client or a direct request to the URL. I think the python client may make it slightly easier to add additional option to the delete request but both a perfectly valid

Reply:
` [1c635a0](/ukaea/piezo/pull/25/commits/1c635a0a1492c8ef09a59bf880936dafce3f3e05)`